src/main/java/edu/kis/powp/command/ComplexCommandFactory.java

```diff
@@ -0,0 +1,52 @@
+package edu.kis.powp.command;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ComplexCommandFactory implements CommandFactory<ComplexCommand> {
+
+    private Job2dDriver job2dDriver;
+    private static List<DriverCommand> commands = new ArrayList<>();
```

Comment:
reusing the same object (static) for different instances does not make sense (and probably leads to errors!)